tests/unit_tests/cli/test_model_factory.py

```diff
@@ -125,6 +125,32 @@ def test_setup_multiple_data_assimilation(poly_case, storage):
     assert "weights" in model._simulation_arguments
 
 
+def test_setup_multiple_data_assimilation_with_restart(poly_case, storage):
+    ert = poly_case
+    args = Namespace(
+        realizations="0-4,7,8",
+        weights="6,4,2",
+        current_case="default",
+        target_case="test_case_%d",
+        start_iteration="1",
+        restart_run=True,
+        prior_ensemble="default",
+    )
+
+    model = model_factory._setup_multiple_data_assimilation(
+        ert,
+        storage,
+        args,
+        UUID(int=0),
+    )
```

Comment:
I just copied the assertions, need to investigate them before expecting this test to pass.